jedeschule/spiders/hessen.py

```diff
@@ -66,26 +97,46 @@ def parse_details(self, response):
             if "http" in link:
                 school["homepage"] = link
 
+        # Extract school type from main content area
         school["schultyp"] = (
             response.xpath('//main//div[@class="col-md-9 col-lg-9"]/text()')
             .extract_first()
             .replace("\n", "")
             .strip()
         )
+        # Extract school ID from URL query parameter
         school["id"] = response.request.url.split("=")[-1]
 
+        # Extract coordinates from OpenStreetMap iframe
+        latitude, longitude = None, None
+        iframe_src = response.xpath('//iframe[contains(@src, "openstreetmap.org")]/@src').get()
+        if iframe_src:
+            latitude, longitude = self._extract_coords_from_osm_url(iframe_src)
+
+        # Filter out placeholder coordinates (-1.0, -1.0) used by Hessen DB for missing data
+        # Example: https://schul-db.bildung.hessen.de/schul_db.html/details/?school_no=9642
+        if latitude == -1.0 and longitude == -1.0:
+            latitude = None
+            longitude = None
+
+        school["latitude"] = latitude
+        school["longitude"] = longitude
+
         yield school
 
     @staticmethod
     def normalize(item: Item) -> School:
+        """Transform raw scraped data into standardized School model"""
         return School(
-            name=item.get("name"),
-            phone=item.get("telefon"),
-            fax=item.get("fax"),
-            website=item.get("homepage"),
-            address=item.get("straße"),
-            city=item.get("ort"),
-            zip=item.get("plz"),
-            school_type=item.get("schultyp"),
-            id="HE-{}".format(item.get("id")),
+            name=item.get("name") or None,
```

Comment:
Why did you add `or None` for all of these? That should be the default, right? (unless you want to explicitly coalesce `""` to `None`?)

Reply:
Addressed all 3 issues.